### ISSUE TITLE
Cannot SelectAndCount when using Join

### ISSUE THREAD

**Issue by jeremyjohn** [go-pg/pg#423]
Example Query:
```
err := db.Model(&sale).
Column("sale.*", "Item", "Item.Stock","Customer" ).  
SelectAndCount()
```
When i Join and then try to SelectAndCount I get this error:  

`ERROR #42803 column "item.id" must appear in the GROUP BY clause or be used in an aggregate function`

**Comment by vmihailenco** [comment]
Can you please provide generated query? Thanks.

Also make sure you are using latest version, because I improved `Count` queries generation last week.

**Comment by jeremyjohn** [comment]
Ok i've updated to the latest version but still unable to get it to work. 
This should explain everything:

**Query:**
```
q := manager.app.DB.Model(&order)
q.Column("order.*", "Project", "ProjectUnit", "ProjectUnit.Status", "CustomerDetail", "OrderStatus", "OrderPaymentMethod")
q.Where("project_unit.id = ?", 3)
q.SelectAndCount()
```

**Terminal Log:**
```
2016/12/01 14:00:33 asm_amd64.s:1998: 
SELECT "order".*,
       "project"."id" AS "project__id",
       "project"."code" AS "project__code",
       "project"."name" AS "project__name",
       "project_unit"."id" AS "project_unit__id",
       "project_unit"."name" AS "project_unit__name",
       "project_unit"."project_id" AS "project_unit__project_id",
       "project_unit"."status_id" AS "project_unit__status_id",
       "project_unit__status"."id" AS "project_unit__status__id",
       "project_unit__status"."name" AS "project_unit__status__name",
       "customer_detail"."id" AS "customer_detail__id",
       "customer_detail"."name" AS "customer_detail__name",
       "order_status"."id" AS "order_status__id",
       "order_status"."name" AS "order_status__name",
       "order_payment_method"."id" AS "order_payment_method__id",
       "order_payment_method"."name" AS "order_payment_method__name",
FROM   "orders" AS "order"
       LEFT OUTER JOIN
       "projects" AS "project"
       ON "project"."id" = "order"."project_id"
       LEFT OUTER JOIN
       "project_units" AS "project_unit"
       ON "project_unit"."id" = "order"."project_unit_id"
       LEFT OUTER JOIN
       "project_unit_statuses" AS "project_unit__status"
       ON "project_unit__status"."id" = "project_unit"."status_id"
       LEFT OUTER JOIN
       "customer_details" AS "customer_detail"
       ON "customer_detail"."id" = "order"."customer_detail_id"
       LEFT OUTER JOIN
       "order_statuses" AS "order_status"
       ON "order_status"."id" = "order"."status_id"
       LEFT OUTER JOIN
       "order_payment_methods" AS "order_payment_method"
       ON "order_payment_method"."id" = "order"."payment_method_id"
WHERE  (project_unit.id = 3)

2016/12/01 14:00:33 asm_amd64.s:1998:
SELECT count(*) FROM "orders" AS "order" WHERE (project_unit.id = 3)

2016/12/01 14:00:33 notifying bugsnag: 
ERROR #42P01 missing FROM-clause entry for table "project_unit" (addr="[::1]:5432")
```

Looks like the Select part is working but the Count part is not doing the Join.

**Comment by vmihailenco** [comment]
Should be fixed in v5.1.3
